src/main/scala/akka/contrib/datareplication/ORMap.scala

```diff
@@ -68,23 +82,63 @@ case class ORMap(
    * INTERNAL API
    */
   private[akka] def put(node: UniqueAddress, key: String, value: ReplicatedData): ORMap =
-    ORMap(keys.add(node, key), values.updated(key, value))
+    if (value.isInstanceOf[ORSet] && values.contains(key))
```

Comment:
I get the feeling here that we're preventing `ORSet` from being `put` just because of `ORMultiMap`. I guess that is ok, but in a general sense, might you want to prevent other types of replicated data collection also from being `put`... or is that too hard... I guess there is no common collection trait.

Perhaps in order to prevent other replicated data collection types we should introduce a new trait that `ORSet` and perhaps others can implement e.g. `PutPreventable` or something much, much better named. Conversely may be replicated data types that can be `put` should mix in a trait. `ORSet` seems very specific.